### ISSUE TITLE
Serve all /api/... requests needed for `pub get` and `pub upgrade` from statically generated files

### ISSUE THREAD

**Issue by sigurdm** [dart-lang/pub-dev#3035]
If we could pregenerate the response to
```
/api/packages/foo
/api/packages/foo.tar.gz
```
And store it for example in a cloud storage bucket, and statically redirect all queries directly to that bucket would make it much easier to feel confident in the availability of `pub get` and `pub upgrade`.

**Comment by jonasfj** [comment]
Not quite, we won't want to redirect. We'll want to serve directly from the bucket...

Thus, we'll need a new subdomain like `get.pub.dev` which hosts this static content.

Note. This is also blocked on popularity metrics being collected by some other means.. hence I think we need a bit of design work first.

**Comment by sigurdm** [comment]
We still want this!
